tests/tests.rs

```diff
@@ -342,21 +342,34 @@ mod foreign_link_test {
     }
 
     #[test]
+    #[cfg(not(has_error_source))]
     fn finds_cause() {
         let chained_error = try_foreign_error().err().unwrap();
         assert_eq!(format!("{}", ForeignErrorCause {}),
                    format!("{}", ::std::error::Error::cause(&chained_error).unwrap()));
     }
 
+    #[test]
+    #[cfg(has_error_source)]
+    fn finds_source() {
+        let chained_error = try_foreign_error().err().unwrap();
+        assert!(::std::error::Error::source(&chained_error).is_none());
+    }
+
     #[test]
     fn iterates() {
         let chained_error = try_foreign_error().err().unwrap();
         let mut error_iter = chained_error.iter();
         assert!(!format!("{:?}", error_iter).is_empty());
         assert_eq!(format!("{}", ForeignError { cause: ForeignErrorCause {} }),
                    format!("{}", error_iter.next().unwrap()));
-        assert_eq!(format!("{}", ForeignErrorCause {}),
-                   format!("{}", error_iter.next().unwrap()));
+        match () {
+            #[cfg(not(has_error_source))]
+            () => assert_eq!(format!("{}", ForeignErrorCause {}),
+                             format!("{}", error_iter.next().unwrap())),
+            #[cfg(has_error_source)]
+            () => assert!(error_iter.next().is_none()),
```

Comment:
Fixed

Reply:
Doing this change broke 1.10 support, so reverting back to my original changes.

Reply:
My bad, attributes on statements is a [1.13.0](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1130-2016-11-10) feature, should have checked minimal supported Rust first.

And just in case I wasn't clear, here I was referring to the assert for `#[cfg(has_error_source)]` being functionally equal to its `#[cfg(not(has_error_source))]` counterpart above.